src/lib/Server/Controller/SessionController.php

```diff
@@ -138,25 +91,34 @@ public function refreshSessionAction($sessionId, Request $request)
             $currentUser,
             $session->getName(),
             $session->getId(),
-            $request->headers->get('X-CSRF-Token'),
+            $this->getCsrfToken(),
             false
         );
     }
 
     /**
-     * @return \Ibexa\Rest\Server\Values\UserSession|\Symfony\Component\HttpFoundation\Response
+     * Refresh given session.
+     *
+     * @deprecated 4.6.7 The "SessionController::refreshSessionAction()" method is deprecated, will be removed in the next API version. Use SessionController::checkSessionAction() instead.
+     *
+     * @throws \Ibexa\Core\Base\Exceptions\UnauthorizedException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
      */
-    public function checkSessionAction(Request $request)
+    public function refreshSessionAction(string $sessionId, Request $request): Values\UserSession|Response
     {
-        $session = $request->getSession();
+        trigger_deprecation(
+            'ibexa/rest',
+            '4.6.7',
+            sprintf('The %s() method is deprecated, will be removed in the next API version.', __METHOD__)
+        );
```

Comment:
These changes are targeted at `5.0.0`. 

```suggestion
        trigger_deprecation(
            'ibexa/rest',
            '5.0.0',
            sprintf('The %s() method is deprecated, will be removed in the next REST API version.', __METHOD__)
        );
```